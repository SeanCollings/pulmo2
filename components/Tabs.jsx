import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';

import { useTheme } from '../hooks/useTheme';

const Tabs = ({ options, onPress, selectedTab }) => {
  const theme = useTheme();
  const length = options.length;

  return (
    <View style={styles.container}>
      {options.map((option, i) => {
        const selected = selectedTab === option;
        return (
          <View
            key={`tab-${i}`}
            style={{
              width: `${100 / length}%`,
            }}
          >
            <TouchableOpacity
              onPress={() => onPress(options[i])}
              activeOpacity={0.8}
              style={{
                ...styles.tabContainer,
                backgroundColor: theme.BACKGROUND,
              }}
            >
              <View style={styles.textContainer}>
                <Text
                  style={{
                    ...styles.tabText,
                    color: theme.TEXT,
                    opacity: theme.DARK ? 0.7 : 0.9,
                  }}
                >
                  {option}
                </Text>
              </View>
              <View
                style={{
                  ...styles.underline,
                  width: `${100 / length}%`,
                  borderBottomColor: selected
                    ? theme.DARK
                      ? theme.DARK_GRAY
                      : theme.BORDER
                    : theme.BACKGROUND,
                }}
              ></View>
            </TouchableOpacity>
          </View>
        );
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
  },
  tabContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 40,
  },
  textContainer: {
    height: '100%',
    justifyContent: 'center',
  },
  tabText: {
    fontFamily: 'tit-regular',
    fontSize: 16,
  },
  underline: {
    borderBottomWidth: 1,
  },
});

export default Tabs;
