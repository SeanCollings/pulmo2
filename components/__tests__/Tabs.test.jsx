import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup, fireEvent } from 'react-native-testing-library';
import Tabs from '../Tabs';

describe('Tabs - unit test', () => {
  let component;
  const options = ['one', 'two'];
  const onPress = jest.fn();
  const selectedTab = 'one';

  afterEach(() => {
    cleanup();
  });

  test('should render', async () => {
    await act(async () => {
      component = create(
        <Tabs options={options} onPress={onPress} selectedTab={selectedTab} />
      );
    });
    expect(component).toMatchSnapshot();
  });

  test('should handle option press', async () => {
    await act(async () => {
      component = create(
        <Tabs options={options} onPress={onPress} selectedTab={selectedTab} />
      );
    });

    const tabButton = component.root.find(
      ({ props }) => props.testID === `tab_${options[0]}`
    );

    fireEvent.press(tabButton);
    expect(onPress).toHaveBeenCalled();
  });
});
