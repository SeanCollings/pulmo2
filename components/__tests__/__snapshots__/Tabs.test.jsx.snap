// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs - unit test should render 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      testID="tab_one"
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "height": "100%",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3e4b57",
                "fontFamily": "tit-regular",
                "fontSize": 16,
                "opacity": 0.9,
              }
            }
          >
            one
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "#CCC",
              "borderBottomWidth": 1,
              "width": "50%",
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
  <View
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      testID="tab_two"
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "height": 40,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "height": "100%",
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#3e4b57",
                "fontFamily": "tit-regular",
                "fontSize": 16,
                "opacity": 0.9,
              }
            }
          >
            two
          </Text>
        </View>
        <View
          style={
            Object {
              "borderBottomColor": "white",
              "borderBottomWidth": 1,
              "width": "50%",
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
